### PR TITLE
Rearrange FullSystemGUI to Save Space

### LINKED ISSUE #1535: Visualizer: Draw field vertically
### Description of the task

It will take up less space overall

### Acceptance criteria

- [ ] As titled

### PR DESCRIPTION
**Pull request by jonathanlew**
### Description
* Rearrange FullSystemGUI to Save Space
* Change dynamic parameter widget for doubles to slider
* Add labels for `Config`s

### Testing Done
manual testing

### Resolved Issues
resolves #1535 (This PR saves space in a different way but achieves the same goals)

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by LiCody — COMMENTED** [review]
Dynamic parameter changes are great, really like the new sliders. The rearrangement of the FullSystemGUI seems to create a lot more wasted space(on the two sides of the field, below the play/tactic info widget) as well as the fixed size of the bottom group of widgets makes it harder to see the field with a smaller window size like 1280 x 720. I personally prefer the old layout but would love to hear what others think about it.

PS. We may want to disable scroll wheeling through values on a parameter slider/spinbox? accidentally caused a SIGABRT while scrolling down

**Comment by akhilveeraghanta** [comment]
@LiCody could you post a screenshot of what it looks like on your computer?

On a 1920x1080p screen it looks like this:
image

**Comment by LiCody** [comment]
> @LiCody could you post a screenshot of what it looks like on your computer?
> 
> On a 1920x1080p screen it looks like this:
> image

Oh nvm, I didn't realize that's the way we wanted to open both, I just had it as a 50/50 split. New layout looks good 👍 .

**Comment by akhilveeraghanta** [comment]
> Oh nvm, I didn't realize that's the way we wanted to open both, I just had it as a 50/50 split. New layout looks good +1 .

This could be very specific to i3, how does it look when you open it like a regular window?

**Comment by LiCody** [comment]
> > Oh nvm, I didn't realize that's the way we wanted to open both, I just had it as a 50/50 split. New layout looks good +1 .
> 
> This could be very specific to i3, how does it look when you open it like a regular window?

image
2OtmE12Oys

Im using Xsrv and their own wm

**Comment by jonathanlew** [comment]
I think that it should be 50/50 split. @LiCody For Xsrv, can you increase the resolution so that the bottom panels are smaller?

**Review by akhilveeraghanta — COMMENTED** [review]
this works really well for me, but will hold off on approving until a couple more people agree with the changes

**Comment by amrm3lm** [comment]
Looks good and more convenient!

**Comment by jonathanlew** [comment]
@LiCody can you try it again? I've managed to make the bottom widgets shorter

**Comment by jonathanlew** [comment]
> PS. We may want to disable scroll wheeling through values on a parameter slider/spinbox? accidentally caused a SIGABRT while scrolling down

that will be fixed in https://github.com/UBC-Thunderbots/Software/issues/1581 after dynamic parameters are done

File: src/software/gui/generic_widgets/dynamic_parameters/dynamic_parameter_widget.cpp [hunk 1]
```diff
@@ -51,18 +50,39 @@ void DynamicParameterWidget::setupParameters(std::shared_ptr<Config> config)
...
     setWidgetResizable(true);
 }
 
+QWidget* DynamicParameterWidget::createConfigLabel(std::shared_ptr<Config> config)
+{
+    QWidget* widget     = new QWidget();
+    QVBoxLayout* layout = new QVBoxLayout(widget);
+
+    QLabel* label = new QLabel(widget);
+    label->setText(QString::fromStdString(config->name()));
```
[the remark this answers is not in the document]
Comment (jonathanlew) [reply src/software/gui/generic_widgets/dynamic_parameters/dynamic_parameter_widget.cpp:75]: done

**Review by akhilveeraghanta — DISMISSED** [review]
:shipit:

**Review by LiCody — APPROVED** [review]
LGTM :tada:

**Review by akhilveeraghanta — APPROVED** [review]

**Pull request merged**